### ISSUE TITLE
IllegalArgumentException `north must be in` when removing last Polyline from folder

### ISSUE THREAD

**Issue by pasniak** [osmdroid/osmdroid#1521]
## Issue Type

[x] Question
[?] Bug

## Description and/or steps/code to reproduce the problem

I have a folder Overlay (modified custom drawing surface by Alex O'Ree) with Polylines.  I've just implemented tap-to-remove-polyline feature:
```
                    val line = PolylineDrawn(map).apply {
                        infoWindow = null
                        color = drawColor
                        setPoints(geoPointsReduced)
                        setOnClickListener {
                            polyline: Polyline, mapView: MapView, geoPoint: GeoPoint ->
                            val isRemoved = folder?.remove(this) == true
                            isRemoved
                        }
                    }
```
and it surprisingly, when **the last** polyline is removed, it fails in `BoundingBox` code! `IllegalArgumentException("north must be in " + tileSystem.toStringLatitudeSpan());`
called from
```
	public boolean remove(Overlay item){
		boolean  b= mOverlayManager.remove(item);
		if(b) recalculateBounds(); //<------- calls the below
		return b;
	}

	private void recalculateBounds() {
		double minLat = Double.MAX_VALUE;
		double minLon = Double.MAX_VALUE;
		double maxLat = -Double.MAX_VALUE;
		double maxLon = -Double.MAX_VALUE;
		for (final Overlay gp : mOverlayManager) {
			BoundingBox box=gp.getBounds();


			minLat = Math.min(minLat, box.getLatSouth());
			minLon = Math.min(minLon,  box.getLonWest());
			maxLat = Math.max(maxLat,  box.getLatNorth());
			maxLon = Math.max(maxLon, box.getLonEast());
		}

		mBounds= new BoundingBox(maxLat, maxLon, minLat, minLon); //<-- bombs here
	}

	public void set(final double north, final double east, final double south, final double west) {
		mLatNorth = north;
		mLonEast = east;
		mLatSouth = south;
		mLonWest = west;
		//validate the values
		//  30 > 0 OK
		// 30 < 0 not ok

        final TileSystem tileSystem = org.osmdroid.views.MapView.getTileSystem();
		if (!tileSystem.isValidLatitude(north))
			throw new IllegalArgumentException("north must be in " + tileSystem.toStringLatitudeSpan()); // N:-1.7976931348623157E308; E:-1.7976931348623157E308; S:1.7976931348623157E308; W:1.7976931348623157E308
```

### Version of osmdroid the issue relates to:

6.1.7-SNAPSHOT

**Comment by monsieurtanuki** [comment]
This is definitely a side-effect bug.
The fastest fix would be to populated min/max lat/lon with the `TileSystem` limit values.

**Comment by monsieurtanuki** [comment]
Fixed by #1524.
